// yet-another-react-lightbox/dist/plugins/Video.js
import * as React from "react";
import PropTypes from "prop-types";
import { SlideTypesPropTypes } from "../types.js";
import { clsx, cssClass, useContainerRect } from "../core/index.js";
SlideTypesPropTypes.push(PropTypes.shape({
    type: PropTypes.oneOf(["video"]).isRequired,
    poster: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number,
    sources: PropTypes.arrayOf(PropTypes.shape({
        src: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
    }).isRequired),
}));
export const VideoSlide = ({ slide: { sources, poster, width, height } }) => {
    const { setContainerRef, containerRect } = useContainerRect();
    const scaleWidthAndHeight = () => {
        if (!width || !height || !containerRect)
            return null;
        const widthBound = width / height > containerRect.width / containerRect.height;
        return {
            width: widthBound ? containerRect.width : Math.round((containerRect.height / height) * width),
            height: !widthBound ? containerRect.height : Math.round((containerRect.width / width) * height),
        };
    };
    return (React.createElement(React.Fragment, null, sources && (React.createElement("div", { ref: setContainerRef, style: {
            width: "100%",
            height: "100%",
        }, className: clsx(cssClass("video_container"), cssClass("flex_center")) }, containerRect && (React.createElement("video", { controls: true, playsInline: true, poster: poster, ...scaleWidthAndHeight() }, sources.map(({ src, type }, index) => (React.createElement("source", { key: index, src: src, type: type })))))))));
};
export const Video = ({ augment }) => {
    augment(({ render: { slide: renderSlide, ...restRender }, ...restProps }) => ({
        render: {
            slide: (slide) => {
                if ("type" in slide && slide.type === "video") {
                    return React.createElement(VideoSlide, { slide: slide });
                }
                return renderSlide === null || renderSlide === void 0 ? void 0 : renderSlide(slide);
            },
            ...restRender,
        },
        ...restProps,
    }));
};
